File: Second year/Design and Programming Environments/TicketManagerJava/src/main/java/Service/EventService.java
package Service;

import Domain.Event;
import Repository.EventRepository;
import Utils.*;

import java.util.ArrayList;
import java.util.List;

public class EventService implements Observable<EventChangeEvent> {

    private List<Observer<EventChangeEvent>> observers=new ArrayList<>();
    private EventRepository eventRepository;

    public EventService(EventRepository eventRepository){
        this.eventRepository=eventRepository;
    }

    public void save(String gameName,Double gamePrice,Integer freeSeats){
        Event event=new Event(gameName,gamePrice,freeSeats);
        eventRepository.save(event);
        notifyObservers(new EventChangeEvent(EventType.ADD,event));
    }

    public void delete(Integer integer){
        eventRepository.delete(integer);
        notifyObservers(new EventChangeEvent(EventType.DELETE,null));
    }

    public void update(Integer integer,String gameName,Double gamePrice,Integer freeSeats){
        Event event=new Event(gameName,gamePrice,freeSeats);
        eventRepository.update(integer,event);
        notifyObservers(new EventChangeEvent(EventType.UPDATE,event));
    }

    public Event findOne(Integer integer){
        return eventRepository.findOne(integer);
    }

    public Integer sizeRepo(){
        return eventRepository.size();
    }

    public Iterable<Event> findAll(){
        return eventRepository.findAll();
    }
    @Override
    public void addObserver(Observer<EventChangeEvent> e) {
        observers.add(e);
    }

    @Override
    public void removeObserver(Observer<EventChangeEvent> e) {
        observers.remove(e);
    }

    @Override
    public void notifyObservers(EventChangeEvent t) {
        observers.forEach(o->o.update(t));
    }
}

File: Second year/Design and Programming Environments/TicketManagerCSharp/TicketManagerCSharp/service/EventService.cs
using System.Collections.Generic;
using System.Linq;
using TicketManagerCSharp.domain;
using TicketManagerCSharp.repository;
using TicketManagerCSharp.utils;

namespace TicketManagerCSharp.service
{
    public class EventService
    {
        private EventRepository eventRepository;
        private List<utils.IObserver<EventChangeEvent>> Observers;

        public EventService(EventRepository eventRepository)
        {
            this.eventRepository = eventRepository;
            Observers = new List<utils.IObserver<EventChangeEvent>>();
        }

        public void AddObserver(utils.IObserver<EventChangeEvent> e)
        {
            Observers.Add(e);
        }

        public void NotifyObservers(EventChangeEvent t)
        {
            Observers.ForEach(x => x.Update(t));
        }

        public void RemoveObserver(utils.IObserver<EventChangeEvent> e)
        {
            Observers.Remove(e);
        }

        public void UpdateMeci(int id, string game_name, double game_price, int free_seats)
        {
            Event ev = new Event(id, game_name, game_price, free_seats);
            eventRepository.Update(id, ev);
            NotifyObservers(new EventChangeEvent(ChangeEventType.UPDATE, ev));
        }

        public Event FindOne(int id)
        {
            return eventRepository.FindOne(id);
        }

        public IList<Event> FindAll()
        {
            return eventRepository.FindAll().ToList();
        }
    }
}